contracts/supplementary/rebalancing-manager/RebalancingTokenManager.sol

```diff
@@ -0,0 +1,456 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.25;
+pragma experimental "ABIEncoderV2";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { AddressArrayUtils } from "../../lib/AddressArrayUtils.sol";
+import { ICore } from "../../core/interfaces/ICore.sol";
+import { IMedian } from "../../external/DappHub/interfaces/IMedian.sol";
+import { IRebalancingSetToken } from "../../core/interfaces/IRebalancingSetToken.sol";
+import { ISetToken } from "../../core/interfaces/ISetToken.sol";
+import { RebalancingHelperLibrary } from "../../core/lib/RebalancingHelperLibrary.sol";
+
+
+/**
+ * @title LinearAuctionPriceCurve
+ * @author Set Protocol
+ *
+ * Contract used to manage a Rebalancing Set Token
+ */
+contract RebalancingTokenManager {
```

Comment:
As a supplementary contract, lets give this a more precise name, it should denote the assets BTCETHRebalancingManager?